server/src/main/java/io/druid/guice/BrokerIntermediateResultsPoolProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.guice;
+
+import io.druid.java.util.common.ISE;
+import io.druid.query.DruidProcessingConfig;
+import io.druid.query.groupby.GroupByQueryConfig;
+import io.druid.query.groupby.strategy.GroupByStrategySelector;
+
+import javax.inject.Inject;
+
+@LazySingleton
+public class BrokerIntermediateResultsPoolProvider extends IntermediateResultsPoolProvider
```

Comment:
I understand why the generic names are confusing. But at the same time, QueryableIndexStorageAdapter is not just used on historicals; it's also used on indexing tasks and on realtime nodes. So naming it HistoricalStorageAdapter is somewhat misleading. For shared classes like that I'd prefer having better docs than naming them for just one of many node types that use them.